packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -240,24 +243,36 @@ export class MediaPlaceholder extends Component {
 			src,
 		} = this.state;
 		return (
-			<div className="editor-media-placeholder__url-input-container block-editor-media-placeholder__url-input-container">
-				<Button
-					className="editor-media-placeholder__button block-editor-media-placeholder__button"
-					onClick={ this.openURLInput }
-					isToggled={ isURLInputVisible }
-					isLarge
-				>
-					{ __( 'Insert from URL' ) }
-				</Button>
-				{ isURLInputVisible && (
-					<InsertFromURLPopover
-						src={ src }
-						onChange={ this.onChangeSrc }
-						onSubmit={ this.onSubmitSrc }
-						onClose={ this.closeURLInput }
-					/>
+			<Fragment>
+				<div className="editor-media-placeholder__url-input-container block-editor-media-placeholder__url-input-container">
+					<Button
+						className="editor-media-placeholder__button block-editor-media-placeholder__button"
+						onClick={ this.openURLInput }
+						isToggled={ isURLInputVisible }
+						isLarge
+					>
+						{ __( 'Insert from URL' ) }
+					</Button>
+					{ isURLInputVisible && (
+						<InsertFromURLPopover
+							src={ src }
+							onChange={ this.onChangeSrc }
+							onSubmit={ this.onSubmitSrc }
+							onClose={ this.closeURLInput }
+						/>
+					) }
+				</div>
+				{ onCancel && (
```

Comment:
This method is `renderUrlSelectionUI`. The cancel button doesn't seem specific to the URL selection UI, so it doesn't feel the right place to declare it. The rule of thumb is that one would expect that when calling `renderUrlSelectionUI` only the URL selection UI is rendered.

I see why it's here though, as this is the only place that is called both when the user does and doesn't have media upload permissions.

Still, I think this will trip up future devs, so would be good to think about ways to improve it.

Reply:
I added a `renderCancelLink` method and it is called in two places just as `renderUrlSelectionUI`. Looks great! :)